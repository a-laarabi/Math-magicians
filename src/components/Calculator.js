import React from 'react';
// import Button from './Button';
import calculate from '../logic/calculate';

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      total: 0,
      next: null,
      operation: null,
    };
  }

    onChangeHandler = (e) => {
      const result = calculate(this.state, e.target.innerText);
      this.setState(result);
    }

    displayResult = (e) => {
      this.setState({
        total: e.target.innerText,
      });
    }

    render() {
      const { total, next, operation } = this.state;
      return (
        <div className="calculator">
          <div className="input" onChange={this.displayResult}>
            {total}
            {operation}
            {next}
          </div>
          <button type="button" className="button AC" onClick={this.onChangeHandler}>AC</button>
          <button type="button" className="button " onClick={this.onChangeHandler}>+/-</button>
          <button type="button" className="button" onClick={this.onChangeHandler}>%</button>
          <button type="button" className="button operator" onClick={this.onChangeHandler}>รท</button>
          <button type="button" className="button" onClick={this.onChangeHandler}>7</button>
          <button type="button" className="button" onClick={this.onChangeHandler}>8</button>
          <button type="button" className="button" onClick={this.onChangeHandler}>9</button>
          <button type="button" className="button operator" onClick={this.onChangeHandler}>x</button>
          <button type="button" className="button" onClick={this.onChangeHandler}>4</button>
          <button type="button" className="button" onClick={this.onChangeHandler}>5</button>
          <button type="button" className="button" onClick={this.onChangeHandler}>6</button>
          <button type="button" className="button operator" onClick={this.onChangeHandler}>-</button>
          <button type="button" className="button" onClick={this.onChangeHandler}>1</button>
          <button type="button" className="button" onClick={this.onChangeHandler}>2</button>
          <button type="button" className="button" onClick={this.onChangeHandler}>3</button>
          <button type="button" className="button operator" onClick={this.onChangeHandler}>+</button>
          <button type="button" className="button bigZero end" onClick={this.onChangeHandler}>0</button>
          <button type="button" className="button end" onClick={this.onChangeHandler}>.</button>
          <button type="button" className="button operator end" onClick={this.onChangeHandler}>=</button>
        </div>
      );
    }
}

export default Calculator;
