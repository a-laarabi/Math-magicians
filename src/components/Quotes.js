import React from 'react';

const Quotes = () => (
  <section className='quote'>
    <p>
      Mathematics is not about numbers, equation, computations or algorithms:
      it is about understanding.- Willium Paul Thurston.
    </p>
  </section>
);

export default Quotes;