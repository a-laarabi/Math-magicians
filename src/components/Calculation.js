import React from 'react';
import Calculator from './Calculator';

const Calculation = () => (
  <section className='claculation'>
    <div className='calculator-title'>
      <h2>Let's do some math!</h2>
    </div>
    <div className='calculator'>
      <Calculator />
    </div>
  </section>
);


export default Calculation;