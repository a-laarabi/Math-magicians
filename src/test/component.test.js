import React from 'react';
import renderer from 'react-test-renderer';
import Calculation from '../components/Calculation';
import Calculator from '../components/calculator';
import Home from '../components/Home';
import Quotes from '../components/Quotes';

describe('test components with snapshot', () => {
  it('test Calculation component with snapshot', () => {
    const tree = renderer.create(<Calculation />);
    expect(tree).toMatchSnapshot();
  });
  it('test Calculator component with snapshot', () => {
    const tree = renderer.create(<Calculator />);
    expect(tree).toMatchSnapshot();
  });
  it('test Home component with snapshot', () => {
    const tree = renderer.create(<Home />);
    expect(tree).toMatchSnapshot();
  });
  it('test Quotes component with snapshot', () => {
    const tree = renderer.create(<Quotes />);
    expect(tree).toMatchSnapshot();
  });

});
