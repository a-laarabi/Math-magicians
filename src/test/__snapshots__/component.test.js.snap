// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test components with snapshot test Calculation component with snapshot 1`] = `
<section
  className="claculation"
>
  <div
    className="calculator-title"
  >
    <h2>
      Let's do some math!
    </h2>
  </div>
  <div
    className="calculator"
  >
    <div
      className="calculator"
    >
      <div
        className="input"
        onChange={[Function]}
      >
        0
      </div>
      <button
        className="button AC"
        onClick={[Function]}
        type="button"
      >
        AC
      </button>
      <button
        className="button "
        onClick={[Function]}
        type="button"
      >
        +/-
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        %
      </button>
      <button
        className="button operator"
        onClick={[Function]}
        type="button"
      >
        รท
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        7
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        8
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        9
      </button>
      <button
        className="button operator"
        onClick={[Function]}
        type="button"
      >
        x
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        4
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        5
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        6
      </button>
      <button
        className="button operator"
        onClick={[Function]}
        type="button"
      >
        -
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="button"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="button operator"
        onClick={[Function]}
        type="button"
      >
        +
      </button>
      <button
        className="button bigZero end"
        onClick={[Function]}
        type="button"
      >
        0
      </button>
      <button
        className="button end"
        onClick={[Function]}
        type="button"
      >
        .
      </button>
      <button
        className="button operator end"
        onClick={[Function]}
        type="button"
      >
        =
      </button>
    </div>
  </div>
</section>
`;

exports[`test components with snapshot test Calculator component with snapshot 1`] = `
<div
  className="calculator"
>
  <div
    className="input"
    onChange={[Function]}
  >
    0
  </div>
  <button
    className="button AC"
    onClick={[Function]}
    type="button"
  >
    AC
  </button>
  <button
    className="button "
    onClick={[Function]}
    type="button"
  >
    +/-
  </button>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    %
  </button>
  <button
    className="button operator"
    onClick={[Function]}
    type="button"
  >
    รท
  </button>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    7
  </button>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    8
  </button>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    9
  </button>
  <button
    className="button operator"
    onClick={[Function]}
    type="button"
  >
    x
  </button>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    4
  </button>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    5
  </button>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    6
  </button>
  <button
    className="button operator"
    onClick={[Function]}
    type="button"
  >
    -
  </button>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    1
  </button>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    2
  </button>
  <button
    className="button"
    onClick={[Function]}
    type="button"
  >
    3
  </button>
  <button
    className="button operator"
    onClick={[Function]}
    type="button"
  >
    +
  </button>
  <button
    className="button bigZero end"
    onClick={[Function]}
    type="button"
  >
    0
  </button>
  <button
    className="button end"
    onClick={[Function]}
    type="button"
  >
    .
  </button>
  <button
    className="button operator end"
    onClick={[Function]}
    type="button"
  >
    =
  </button>
</div>
`;

exports[`test components with snapshot test Home component with snapshot 1`] = `
<section
  className="home"
>
  <h2>
    Welcome to our page!
  </h2>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
  </p>
  <p>
    Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat.
  </p>
</section>
`;

exports[`test components with snapshot test Quotes component with snapshot 1`] = `
<section
  className="quote"
>
  <p>
    Mathematics is not about numbers, equation, computations or algorithms: it is about understanding.- Willium Paul Thurston.
  </p>
</section>
`;
